File: tests/javascript/two_pointers.test.js
const fs = require('fs');
const path = require('path');

// Load implementation
const algoPath = path.join(__dirname, '../../two_pointers/two-pointers.js');
const code = fs.readFileSync(algoPath, 'utf8');
eval(code);

describe('Two_Pointers', () => {
  test('test case 1', () => {
    expect(two_sum([[2, 7, 11, 15]], 9)).toBe([0, 1]);
  });

  test('test case 2', () => {
    expect(two_sum([[3, 2, 4]], 6)).toBe([1, 2]);
  });

});

File: tests/java/TwoPointersTest.java
public class TwoPointersTest {
    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5};
        int[] result = TwoPointers.twoSum(arr, 9);
        assert result[0] == 3 && result[1] == 4 : "Two pointers failed";
        System.out.println("TwoPointersTest passed");
    }
}
